docs/development/concepts/hotwire-view-components/README.md

```diff
@@ -0,0 +1,275 @@
+      )
+      # Note: the target_component does not get rendered
+      # the instatiation is just required for the turbo-stream generation
+
+      # you can use multiple turbo_stream methods in one controller action
+      # e.g. update the new component to render an initial form
+      update_via_turbo_stream(
+        component: JournalNewComponent.new(work_package: journal.work_package)
+      )
+    else
+      # optionally set a turbo status for the response
+      @turbo_status = :bad_request
+
+      # trigger a flash message via turbo-stream
+      # more on this here lookbook/pages/patterns/flash_banner
+      update_flash_message_via_turbo_stream(
+        message: journal.errors.full_messages.join(", "),
+        scheme: :danger
+      )
+    end
+
+    # respond with turbo streams which were collected in the @turbo_streams variable behind the scenes
+    # handy if this method is just meant to respond to turbo-stream requests
+    respond_with_turbo_streams
+  end
+
+  # ...
+end
+```
+
+## Mixing turbo-streams and other responses
+
+TODO: Discuss the below example
+
+```ruby
+class JournalController < ApplicationController
+  include OpTurbo::ComponentStream # include this module!
+
+  # ...
+
+  def update
+    # ...
+
+    respond_to do |format|
+      format.html do
+        # ...
+      end
+      format.turbo_stream do
+        update_via_turbo_stream(
+          component: JournalShowComponent.new(journal: journal)
+        )
+
+        render turbo_stream: turbo_streams, status: :ok
+      end
+    end
+  end
+
+  # ...
+end
+```
+
+## Usage alongside Primer Forms/Buttons
+
+TODO: is `turbo: true` required here?
+
+```ruby
+<%=
+  component_wrapper do
+    # ...  
+    primer_form_with(
+      model: journal,
+      method: :put,
+      data: { turbo: true, turbo_stream: true }, # add this!
+      url: journal_path(id: journal.id)
+    ) do |f|
+      # ...
+    end
+    # ...
+  end
+%>
+```
+
+```ruby
+<%=
+  component_wrapper do
+    # ...  
+    render(Primer::Beta::Button.new(
+      scheme: :secondary,
+      size: :medium,
+      tag: :a,
+      href: cancel_edit_journal_path(journal.id),
+      data: { turbo: true, turbo_stream: true } # add this!
+    )) do
+      t("button_cancel")
+    end
+    # ...
+  end
+%>
+```
+
+## Requesting turbo-streams within Stimulus controllers
+
+If for some reason you need to request a turbo-stream programmatically from within a Stimulus controller, you can use the `TurboRequestsService` to do so.
+
+TODO: Discuss the TurboRequestsService API
+
+```typescript
+import { Controller } from '@hotwired/stimulus';
+import { TurboRequestsService } from 'core-app/core/turbo/turbo-requests.service';
+
+export default class IndexController extends Controller {
+
+  private turboRequests:TurboRequestsService;
+
+  async connect() {
+    const context = await window.OpenProject.getPluginContext();
+    this.turboRequests = context.services.turboRequests;
```

Comment:
Sounds magic :)
Is `TurboRequestsService` an angular service? I guess it does not really matter as long as it works.

Reply:
yes it's an Angular service https://github.com/opf/openproject/blob/3d0401e8b22c484734cad96a57fe9738df5e3f0c/frontend/src/app/core/turbo/turbo-requests.service.ts#L5